### ISSUE TITLE
Crash in pipeline handler console spam

### ISSUE THREAD

**Issue by vadimyer** [OpenMods/OpenBlocks#791]
Hello,

I have this kind of spam in console, server is Thermos B55:

```
[12:16:19] [Server thread/ERROR]: Crash in pipeline handler
java.lang.IllegalStateException: Failed to select and send message (selector openmods.network.targets.SelectEntityWatchers@5d71fcd6, arg: cpw.mods.fml.common.network.handshake.NetworkDispatcher@5f48d738, channel: OpenMods|E, side: SERVER)
	at openmods.network.ExtendedOutboundHandler.write(ExtendedOutboundHandler.java:57) ~[ExtendedOutboundHandler.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeWrite(DefaultChannelHandlerContext.java:644) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.write(DefaultChannelHandlerContext.java:698) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.write(DefaultChannelHandlerContext.java:637) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:115) ~[MessageToMessageEncoder.class:?]
	at io.netty.handler.codec.MessageToMessageCodec.write(MessageToMessageCodec.java:116) ~[MessageToMessageCodec.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeWrite(DefaultChannelHandlerContext.java:644) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.write(DefaultChannelHandlerContext.java:698) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.writeAndFlush(DefaultChannelHandlerContext.java:688) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.writeAndFlush(DefaultChannelHandlerContext.java:717) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:893) ~[DefaultChannelPipeline.class:?]
	at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:239) ~[AbstractChannel.class:?]
	at openmods.network.senders.PacketSenderBase.sendMessage(PacketSenderBase.java:22) [PacketSenderBase.class:?]
	at openmods.network.event.NetworkEvent.sendToAll(NetworkEvent.java:37) [NetworkEvent.class:?]
	at openblocks.common.MapDataManager.sendUpdates(MapDataManager.java:185) [MapDataManager.class:?]
	at openblocks.common.ServerTickHandler.onServerTick(ServerTickHandler.java:23) [ServerTickHandler.class:?]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler_281_ServerTickHandler_onServerTick_ServerTickEvent.invoke(.dynamic) [?:1.7.10-1614.55]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54) [ASMEventHandler.class:1.7.10-1614.55]
	at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:140) [EventBus.class:1.7.10-1614.55]
	at cpw.mods.fml.common.FMLCommonHandler.onPreServerTick(FMLCommonHandler.java:264) [FMLCommonHandler.class:1.7.10-1614.55]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:819) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:683) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
Caused by: java.lang.IllegalArgumentException: Arg must be Entity
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:125) ~[server-1.7.10.jar:?]
	at openmods.network.targets.SelectEntityWatchers.listDispatchers(SelectEntityWatchers.java:28) ~[SelectEntityWatchers.class:?]
	at openmods.network.ExtendedOutboundHandler.write(ExtendedOutboundHandler.java:50) ~[ExtendedOutboundHandler.class:?]
	... 22 more
[12:16:20] [Server thread/INFO]: FallenAnGeL issued server command: /we cui
[12:16:20] [Server thread/ERROR]: Crash in pipeline handler
java.lang.IllegalStateException: Failed to select and send message (selector openmods.network.targets.SelectEntityWatchers@5d71fcd6, arg: cpw.mods.fml.common.network.handshake.NetworkDispatcher@5f48d738, channel: OpenMods|E, side: SERVER)
	at openmods.network.ExtendedOutboundHandler.write(ExtendedOutboundHandler.java:57) ~[ExtendedOutboundHandler.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeWrite(DefaultChannelHandlerContext.java:644) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.write(DefaultChannelHandlerContext.java:698) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.write(DefaultChannelHandlerContext.java:637) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:115) ~[MessageToMessageEncoder.class:?]
	at io.netty.handler.codec.MessageToMessageCodec.write(MessageToMessageCodec.java:116) ~[MessageToMessageCodec.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeWrite(DefaultChannelHandlerContext.java:644) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.write(DefaultChannelHandlerContext.java:698) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.writeAndFlush(DefaultChannelHandlerContext.java:688) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.writeAndFlush(DefaultChannelHandlerContext.java:717) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:893) ~[DefaultChannelPipeline.class:?]
	at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:239) ~[AbstractChannel.class:?]
	at openmods.network.senders.PacketSenderBase.sendMessage(PacketSenderBase.java:22) [PacketSenderBase.class:?]
	at openmods.network.event.NetworkEvent.sendToAll(NetworkEvent.java:37) [NetworkEvent.class:?]
	at openblocks.common.MapDataManager.sendUpdates(MapDataManager.java:185) [MapDataManager.class:?]
	at openblocks.common.ServerTickHandler.onServerTick(ServerTickHandler.java:23) [ServerTickHandler.class:?]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler_281_ServerTickHandler_onServerTick_ServerTickEvent.invoke(.dynamic) [?:1.7.10-1614.55]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54) [ASMEventHandler.class:1.7.10-1614.55]
	at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:140) [EventBus.class:1.7.10-1614.55]
	at cpw.mods.fml.common.FMLCommonHandler.onPreServerTick(FMLCommonHandler.java:264) [FMLCommonHandler.class:1.7.10-1614.55]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:819) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:683) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
Caused by: java.lang.IllegalArgumentException: Arg must be Entity
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:125) ~[server-1.7.10.jar:?]
	at openmods.network.targets.SelectEntityWatchers.listDispatchers(SelectEntityWatchers.java:28) ~[SelectEntityWatchers.class:?]
	at openmods.network.ExtendedOutboundHandler.write(ExtendedOutboundHandler.java:50) ~[ExtendedOutboundHandler.class:?]
	... 22 more
[12:16:20] [Server thread/ERROR]: Crash in pipeline handler
java.lang.IllegalStateException: Failed to select and send message (selector openmods.network.targets.SelectEntityWatchers@5d71fcd6, arg: cpw.mods.fml.common.network.handshake.NetworkDispatcher@5f48d738, channel: OpenMods|E, side: SERVER)
	at openmods.network.ExtendedOutboundHandler.write(ExtendedOutboundHandler.java:57) ~[ExtendedOutboundHandler.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeWrite(DefaultChannelHandlerContext.java:644) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.write(DefaultChannelHandlerContext.java:698) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.write(DefaultChannelHandlerContext.java:637) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:115) ~[MessageToMessageEncoder.class:?]
	at io.netty.handler.codec.MessageToMessageCodec.write(MessageToMessageCodec.java:116) ~[MessageToMessageCodec.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeWrite(DefaultChannelHandlerContext.java:644) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.write(DefaultChannelHandlerContext.java:698) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.writeAndFlush(DefaultChannelHandlerContext.java:688) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.writeAndFlush(DefaultChannelHandlerContext.java:717) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:893) ~[DefaultChannelPipeline.class:?]
	at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:239) ~[AbstractChannel.class:?]
	at openmods.network.senders.PacketSenderBase.sendMessage(PacketSenderBase.java:22) [PacketSenderBase.class:?]
	at openmods.network.event.NetworkEvent.sendToAll(NetworkEvent.java:37) [NetworkEvent.class:?]
	at openblocks.common.MapDataManager.sendUpdates(MapDataManager.java:185) [MapDataManager.class:?]
	at openblocks.common.ServerTickHandler.onServerTick(ServerTickHandler.java:23) [ServerTickHandler.class:?]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler_281_ServerTickHandler_onServerTick_ServerTickEvent.invoke(.dynamic) [?:1.7.10-1614.55]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54) [ASMEventHandler.class:1.7.10-1614.55]
	at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:140) [EventBus.class:1.7.10-1614.55]
	at cpw.mods.fml.common.FMLCommonHandler.onPreServerTick(FMLCommonHandler.java:264) [FMLCommonHandler.class:1.7.10-1614.55]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:819) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:683) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
Caused by: java.lang.IllegalArgumentException: Arg must be Entity
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:125) ~[server-1.7.10.jar:?]
	at openmods.network.targets.SelectEntityWatchers.listDispatchers(SelectEntityWatchers.java:28) ~[SelectEntityWatchers.class:?]
	at openmods.network.ExtendedOutboundHandler.write(ExtendedOutboundHandler.java:50) ~[ExtendedOutboundHandler.class:?]
	... 22 more
[12:16:21] [Server thread/ERROR]: Crash in pipeline handler
java.lang.IllegalStateException: Failed to select and send message (selector openmods.network.targets.SelectEntityWatchers@5d71fcd6, arg: cpw.mods.fml.common.network.handshake.NetworkDispatcher@5f48d738, channel: OpenMods|E, side: SERVER)
	at openmods.network.ExtendedOutboundHandler.write(ExtendedOutboundHandler.java:57) ~[ExtendedOutboundHandler.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeWrite(DefaultChannelHandlerContext.java:644) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.write(DefaultChannelHandlerContext.java:698) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.write(DefaultChannelHandlerContext.java:637) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:115) ~[MessageToMessageEncoder.class:?]
	at io.netty.handler.codec.MessageToMessageCodec.write(MessageToMessageCodec.java:116) ~[MessageToMessageCodec.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.invokeWrite(DefaultChannelHandlerContext.java:644) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.write(DefaultChannelHandlerContext.java:698) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.writeAndFlush(DefaultChannelHandlerContext.java:688) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelHandlerContext.writeAndFlush(DefaultChannelHandlerContext.java:717) ~[DefaultChannelHandlerContext.class:?]
	at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:893) ~[DefaultChannelPipeline.class:?]
	at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:239) ~[AbstractChannel.class:?]
	at openmods.network.senders.PacketSenderBase.sendMessage(PacketSenderBase.java:22) [PacketSenderBase.class:?]
	at openmods.network.event.NetworkEvent.sendToAll(NetworkEvent.java:37) [NetworkEvent.class:?]
	at openblocks.common.MapDataManager.sendUpdates(MapDataManager.java:185) [MapDataManager.class:?]
	at openblocks.common.ServerTickHandler.onServerTick(ServerTickHandler.java:23) [ServerTickHandler.class:?]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler_281_ServerTickHandler_onServerTick_ServerTickEvent.invoke(.dynamic) [?:1.7.10-1614.55]
	at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54) [ASMEventHandler.class:1.7.10-1614.55]
	at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:140) [EventBus.class:1.7.10-1614.55]
	at cpw.mods.fml.common.FMLCommonHandler.onPreServerTick(FMLCommonHandler.java:264) [FMLCommonHandler.class:1.7.10-1614.55]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:819) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:683) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
Caused by: java.lang.IllegalArgumentException: Arg must be Entity
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:125) ~[server-1.7.10.jar:?]
	at openmods.network.targets.SelectEntityWatchers.listDispatchers(SelectEntityWatchers.java:28) ~[SelectEntityWatchers.class:?]
	at openmods.network.ExtendedOutboundHandler.write(ExtendedOutboundHandler.java:50) ~[ExtendedOutboundHandler.class:?]
	... 22 more
```

ATM that's all about it. I'm trying to find out what's causing that and if that crashes the client as well, but no crash reports were reported to me, or if this is the cause of some lag.

**Comment by boq** [comment]
This shouldn't affect clients. I guess server performance could be degraded if that happened every tick.

Anyway, this is part of height map code (triggered by update of one of those), but I'm not really sure how this can happen (probably some concurrency issue).

How often does it happen? Can you reliably reproduce it after server restart?

**Comment by boq** [comment]
Fixed in OpenMods/OpenModsLib@9fafa7b
